### PR TITLE
docs: Updates troubleshoot guide to show how to extract `ErrorInfo` from `RpcException`.

### PR DESCRIPTION
**Pull request by anuragsrivstv**
- Updates troubleshoot file to show how to extract ErrorInfo from RpcException.

- Includes a sample to demo how to use RpcException property of SpannerException.

### CODE REVIEW

File: docs/devsite-help/troubleshooting.md
Comment (hemanshv) [inline docs/devsite-help/troubleshooting.md:246]: nit: multiple space before ex

**Comment by hemanshv** [comment]
Added comments for 2 minor nits.
LGTM

File: docs/devsite-help/troubleshooting.md
Comment (Rishabh-V) [reply docs/devsite-help/troubleshooting.md:246]: yep. Should be just one whitespace.

File: docs/devsite-help/troubleshooting.md
Comment (Rishabh-V) [inline docs/devsite-help/troubleshooting.md:253]: What remedial action?
Only reason I say this is that without concrete examples, this may paint an incomplete picture and the reader may end up being curious about it with no link or reference to the additional information. I 'll let Jon decide on it so please don't make any changes for now.

File: docs/devsite-help/troubleshooting.md
Comment (Rishabh-V) [inline docs/devsite-help/troubleshooting.md:257]: Given the changes have been pushed recently and not released yet, we *may* want to release those changes and then say that "starting version 4.X.X or higher", so that users of existing versions of library understand they need to upgrade and this doesn't create confusion.

File: docs/devsite-help/troubleshooting.md
Comment (Rishabh-V) [inline docs/devsite-help/troubleshooting.md:261]: Maybe 
```suggestion
value may be null if the `SpannerException` was not caused by an `RpcException`.
```

File: docs/devsite-help/troubleshooting.md
Comment (Rishabh-V) [inline docs/devsite-help/troubleshooting.md:269]: nit: `.` at the end of comment.

File: docs/devsite-help/troubleshooting.md
Comment (Rishabh-V) [inline docs/devsite-help/troubleshooting.md:274]: missing space between `if` and `(`
```suggestion
   if (rpcException != null)
```

Comment (Rishabh-V) [reply docs/devsite-help/troubleshooting.md:274]: Also, I see in https://github.com/googleapis/google-cloud-dotnet/pull/9484#pullrequestreview-1234636109 that Jon mentions that
```
I would definitely suggest changing this to be focused on RpcException. We can have one example showing that in
detail. We can have one example showing that in detail... then a subsection for SpannerException, with a brief
example using the pattern match (and maybe having another catch clause for SpannerException when there isn't an
inner RpcException?) that refers to the earlier example.
```
so note on `SpannerException`  should be a subsection of the above `RpcException` section instead of a new section (I may be wrong, but wanted to check and be sure)

**Review by Rishabh-V — COMMENTED** [review]
Few minor nits. Please have a look and request Jon's review to finalize the changes. Thank you.

File: docs/devsite-help/troubleshooting.md
Comment (anuragsrivstv) [reply docs/devsite-help/troubleshooting.md:257]: Make sense. So should we wait for its release before merging this PR ?

File: docs/devsite-help/troubleshooting.md
Comment (jskeet) [inline docs/devsite-help/troubleshooting.md:238]: You need to specify the way in which this is constant. (I understand this text is taken from error_details.proto, but we can clarify here.)

I'd also avoid the "It is" wording here and below. Here's a potential rewriting:

```suggestion
- `Reason`: the proximate cause of the error. This is a stable unique identifier for this
   type of error within the domain, and can be used to programmatically determine
   how to handle the error.
- `Domain`: the logical grouping to which the error belongs. The error domain
   is typically the registered service name of the tool or product that
   generates the error. Example: "pubsub.googleapis.com". If the error 
   is generated by some common infrastructure, the error domain will be 
   a globally unique value that identifies the infrastructure. 
For Google API infrastructure, the error domain is "googleapis.com".
- `Metadata`: additional structured details about the error.
```

File: docs/devsite-help/troubleshooting.md
Comment (jskeet) [inline docs/devsite-help/troubleshooting.md:244]:
```suggestion
    // Code that can throw the RpcException.
```

File: docs/devsite-help/troubleshooting.md
Comment (jskeet) [inline docs/devsite-help/troubleshooting.md:251]:
```suggestion
    if (ex.GetErrorInfo() is ErrorInfo errorInfo)
    {
        var domain = errorInfo.Domain;
        var reason = errorInfo.Reason;
        var metadata = errorInfo.Metadata;
        // ...
    }
```

(It's entirely possible for `ex.GetErrorInfo()` to return null.

Comment (jskeet) [reply docs/devsite-help/troubleshooting.md:251]: We may want to have a comment about what the user can still do if there isn't an ErrorInfo, mind you. (They still have the status and the message.)

File: docs/devsite-help/troubleshooting.md
Comment (jskeet) [reply docs/devsite-help/troubleshooting.md:253]: That's really application-specific and API-specific. I think it's fine to leave the comment as is (within the suggestion above).

File: docs/devsite-help/troubleshooting.md
Comment (jskeet) [reply docs/devsite-help/troubleshooting.md:257]: Yes, or split the PR into two: one just for RpcException and one for SpannerException.
Alternatively, given that this is completely Spanner-specific, we could move it into the Spanner docs - and merge that now, so that it'll be published in the next Spanner release. I think that would be better.

Comment (anuragsrivstv) [reply docs/devsite-help/troubleshooting.md:257]: Sure. I'll remove SpannerException section from troubleshoot.md and include it in Spanner docs in a separate PR once the help page with RpcException usage is published.

File: docs/devsite-help/troubleshooting.md
Comment (anuragsrivstv) [reply docs/devsite-help/troubleshooting.md:238]: Incorporated changes.

File: docs/devsite-help/troubleshooting.md
Comment (anuragsrivstv) [reply docs/devsite-help/troubleshooting.md:251]: Sure.

File: docs/devsite-help/troubleshooting.md
Comment (jskeet) [inline docs/devsite-help/troubleshooting.md:239]: This example still has unconventional indentation - it looks like it's three spaces for the first indent, and two for the second.

File: docs/devsite-help/troubleshooting.md
Comment (jskeet) [inline docs/devsite-help/troubleshooting.md:256]:
```suggestion
      // Use RpcException's other properties (e.g. Status and Message) to handle the exception appropriately
```

File: docs/devsite-help/troubleshooting.md
Comment (anuragsrivstv) [reply docs/devsite-help/troubleshooting.md:239]: Reformatted the indentation.

**Review by jskeet — APPROVED** [review]

**Comment by anuragsrivstv** [comment]
Merging now.

**Pull request merged**
